smack-extensions/src/main/java/org/jivesoftware/smackx/caps2/Caps2Manager.java

```diff
@@ -71,32 +72,27 @@ private Caps2Manager(XMPPConnection connection) {
         if (autoEnable) {
             publishSupportForECaps2();
         }
-
-        connection.addConnectionListener(new ConnectionListener() {
-            @Override
-            public void connected(XMPPConnection connection) {
-                 processCapsStreamFeatureIfAvailable(connection);
-            }
-            private void processCapsStreamFeatureIfAvailable(XMPPConnection connection) {
-                Caps2Element caps2Element = connection.getFeature(Caps2Element.class);
-                if (caps2Element == null) {
-                    return;
-                }
-                DomainBareJid from = connection.getXMPPServiceDomain();
-                addCapsExtensionInfo(from, caps2Element);
-            }
-        });
     }
 
     public void publishSupportForECaps2() {
         ServiceDiscoveryManager sdm = ServiceDiscoveryManager.getInstanceFor(connection());
         sdm.addFeature(Caps2Element.NAMESPACE);
-        Logger.getAnonymousLogger().info("adiaholic : caps2 feature added");
     }
 
-    private static void addCapsExtensionInfo(DomainBareJid from, Caps2Element caps2Element) {
-        from.asBareJid();
-        caps2Element.getClass();
+    public void publishEntityCapabilities() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, UnsupportedEncodingException, NoSuchAlgorithmException {
+        ServiceDiscoveryManager sdm = ServiceDiscoveryManager.getInstanceFor(connection());
+        DiscoverInfo discoverInfo = sdm.discoverInfo(connection().getUser());
+
+        List<String> algoList = new ArrayList<String>();
```

Comment:
This could be `Collections.singletonList(defaultAlgo)`, or does the list need to be mutable?